src/main/java/com/artipie/rpm/Repomd.java

```diff
@@ -119,71 +119,95 @@ private Completable loadRepomd(final Path file) {
-                    .registerNs("ns", "http://linux.duke.edu/metadata/repo");
-                final List<XML> nodes = xml.nodes(
-                    String.format("/ns:repomd/data[type='%s']", type)
-                );
-                final RxStorageWrapper rxsto = new RxStorageWrapper(this.storage);
-                final Completable res = this.metaFile(nodes, file);
-                final Path gzip = Files.createTempFile("x", ".gz");
-                return res.andThen(
-                    SingleInterop.fromFuture(this.policy.name(type, new RxFile(file).flow()))
-                ).map(target -> String.format("repodata/%s.xml", target))
-                    .flatMapCompletable(
-                        key -> act.update(file)
-                            .andThen(Repomd.gzip(file, gzip))
-                            .andThen(
-                                Single.just(
-                                    new RepoXml()
-                                ).doOnSuccess(
-                                    rxml -> rxml.base(type, key)
-                                ).zipWith(
-                                    Single.fromCallable(() -> Files.size(file)),
-                                    RepoXml::openSize
-                                ).zipWith(
-                                    Single.fromCallable(() -> Files.size(gzip)),
-                                    RepoXml::size
-                                ).flatMap(
-                                    rxml -> new Checksum(gzip).sha().map(rxml::checksum)
-                                ).zipWith(
-                                    new Checksum(file).sha(),
-                                    RepoXml::openChecksum
-                                ).map(RepoXml::timestamp)
-                            )
-                            .flatMapCompletable(
-                                rxml ->
-                                    rxsto.save(new Key.From(key), new RxFile(file).flow())
-                                        .andThen(
-                                            rxsto.save(
-                                                new Key.From(String.format("%s.gz", key)),
-                                                new RxFile(gzip).flow()
-                                            )
-                                        )
-                                        .andThen(new Update(temp).apply(rxml))
-                                        .andThen(
-                                            Completable.fromAction(
-                                                () -> new BlockingStorage(this.storage).save(
-                                                    new Key.From("repodata/repomd.xml"),
-                                                    Files.readAllBytes(temp)
-                                                )
+        return Single.fromCallable(() -> Files.createTempFile(type, ".xml"))
```

Comment:
@g4s8 I think that you could leave an issue to fix this methods / architecture: currently both `performUpdate` methods do a lot of thinks (creating files, zipping contents, saving file) and I think that these behaviors should be split into several methods